### PR TITLE
tests: Migrate and combine ENV tests for `*_INET_PROTOCOLS`

### PR DESCRIPTION
**Pull request by polarathene**
# Description

These two features + tests were introduced years apart but serve the same functionality, just for either Postfix or Dovecot. Merged into single test.

Original PRs:

- [`DOVECOT_INET_PROTOCOLS`](https://github.com/docker-mailserver/docker-mailserver/pull/2358)
- [`POSTFIX_INET_PROTOCOLS`](https://github.com/docker-mailserver/docker-mailserver/pull/1505)

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: test/tests/parallel/set1/network_inet_protocols.bats [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  _init_with_defaults
+  _common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+
+  _run_in_container grep '^#listen =' /etc/dovecot/dovecot.conf
+  assert_success
+  assert_output '#listen = *, ::'
+
+  _run_in_container postconf inet_protocols
+  assert_output 'inet_protocols = all'
+  assert_success
```
Comment (georglauterbach) [inline test/tests/parallel/set1/network_inet_protocols.bats:37]: probably better to assert success before asserting the output, but should work both ways. Checking whether the command succeeded in the first place should be preferred though.

Comment (georglauterbach) [inline test/tests/parallel/set1/network_inet_protocols.bats:37]: Probably better to assert success before asserting the output, but should work both ways. Checking whether the command succeeded in the first place should be preferred though.

File: test/tests/parallel/set1/network_inet_protocols.bats [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  _init_with_defaults
+  _common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+
+  _run_in_container grep '^listen =' /etc/dovecot/dovecot.conf
+  assert_success
+  assert_output 'listen = *'
+
+  _run_in_container postconf inet_protocols
+  assert_output 'inet_protocols = ipv4'
+  assert_success
```
Comment (georglauterbach) [inline test/tests/parallel/set1/network_inet_protocols.bats:56]: Here as well (see comment above).

**Review by georglauterbach — DISMISSED** [review]
Two suggestions notes, LGTM 👍🏼

File: test/tests/parallel/set1/network_inet_protocols.bats (hunk 2 above)
Comment (polarathene) [inline test/tests/parallel/set1/network_inet_protocols.bats:56]:
```suggestion
  assert_success
  assert_output 'inet_protocols = ipv4'
```

File: test/tests/parallel/set1/network_inet_protocols.bats [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  _init_with_defaults
+  _common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+
+  _run_in_container grep '^listen =' /etc/dovecot/dovecot.conf
+  assert_success
+  assert_output 'listen = [::]'
+
+  _run_in_container postconf inet_protocols
+  assert_output 'inet_protocols = ipv6'
+  assert_success
```
Comment (polarathene) [inline test/tests/parallel/set1/network_inet_protocols.bats:75]:
```suggestion
  assert_success
  assert_output 'inet_protocols = ipv6'
```

**Review by georglauterbach — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Pull request merged**